Makefile

```diff
@@ -15,6 +15,36 @@ ut: update-version
 	calico/test \
 	nosetests tests/unit -c nose.cfg
 
+# Run system tests.
+st: docker-image run-etcd run-k8s-apiserver
+	./tests/system/apiserver-reconnection.sh
+	$(MAKE) stop-k8s-apiserver stop-etcd
+
+GET_CONTAINER_IP := docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}'
+K8S_VERSION=1.5.3
+.PHONY: run-k8s-apiserver stop-k8s-apiserver run-etcd stop-etcd
+run-k8s-apiserver: stop-k8s-apiserver
+	ETCD_IP=`$(GET_CONTAINER_IP) st-etcd` && \
+	docker run --detach \
+	  --name st-apiserver \
+	gcr.io/google_containers/hyperkube-amd64:v$(K8S_VERSION) \
+		  /hyperkube apiserver --etcd-servers=http://$${ETCD_IP}:2379 \
+		  --service-cluster-ip-range=10.101.0.0/16 -v=10
+
+stop-k8s-apiserver:
+	@-docker rm -f st-apiserver
+	sleep 2
+
+run-etcd: stop-etcd
+	docker run --detach \
+	--name st-etcd quay.io/coreos/etcd \
```

Comment:
I suspect we probably want to pin the version of etcd we're using for consistency.  The most recent v3 release should do.